redisearch/client_test.go

```diff
@@ -475,48 +489,58 @@ func TestClient_GetTagVals(t *testing.T) {
 	tags, err = c.GetTagVals("notexit", "tags")
 	assert.NotNil(t, err)
 	assert.Nil(t, tags)
+	teardown(c)
 }
 
 func TestClient_SynAdd(t *testing.T) {
 	c := createClient("testsynadd")
-
-	sc := NewSchema(DefaultOptions).
-		AddField(NewTextField("name")).
-		AddField(NewTextField("addr"))
-	c.Drop()
-	err := c.CreateIndex(sc)
-	assert.Nil(t, err)
-
-	gid, err := c.SynAdd("testsynadd", []string{"girl", "baby"})
-	assert.Nil(t, err)
-	assert.True(t, gid >= 0)
-	ret, err := c.SynUpdate("testsynadd", gid, []string{"girl", "baby"})
+	version, err := c.GetRediSearchVersion()
 	assert.Nil(t, err)
-	assert.Equal(t, "OK", ret)
+	if version <= 10699 {
+		sc := NewSchema(DefaultOptions).
+			AddField(NewTextField("name")).
+			AddField(NewTextField("addr"))
+		c.Drop()
+		err := c.CreateIndex(sc)
+		assert.Nil(t, err)
+
+		gid, err := c.SynAdd("testsynadd", []string{"girl", "baby"})
+		assert.Nil(t, err)
+		assert.True(t, gid >= 0)
+		ret, err := c.SynUpdate("testsynadd", gid, []string{"girl", "baby"})
+		assert.Nil(t, err)
+		assert.Equal(t, "OK", ret)
+	}
+	teardown(c)
 }
 
 func TestClient_SynDump(t *testing.T) {
 	c := createClient("testsyndump")
-
-	sc := NewSchema(DefaultOptions).
-		AddField(NewTextField("name")).
-		AddField(NewTextField("addr"))
-	c.Drop()
-	err := c.CreateIndex(sc)
+	version, err := c.GetRediSearchVersion()
 	assert.Nil(t, err)
+	if version <= 10699 {
```

Comment:
1.8

Reply:
And also here, syndump should also work, only the format was changed.

Reply:
addressed in the latest commit.